File: backend/src/main/java/um/edu/ar/backend/infrastructure/web/dto/EventoDetalleResponse.java
package um.edu.ar.backend.infrastructure.web.dto;


import java.time.Instant;
import java.util.List;

public record EventoDetalleResponse(
        String titulo,
        String resumen,
        String descripcion,
        Instant fecha,
        String direccion,
        String imagen,
        int filaAsientos,
        int columnAsientos,
        double precioEntrada,
        EventoTipoDto eventoTipo,
        List<IntegranteDto> integrantes,
        Long id
) {
    public record EventoTipoDto(String nombre, String descripcion) {}
    public record IntegranteDto(String nombre, String apellido, String identificacion) {}
}

File: cliente_movil/composeApp/src/commonMain/kotlin/um/edu/ar/model/EventoDetalleResponse.kt
package um.edu.ar.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class EventoDetalleResponse(
    val titulo: String,
    val resumen: String,
    val descripcion: String,
    val fecha: String,
    val direccion: String,
    val imagen: String? = null,
    @SerialName("filaAsientos")
    val filaAsientos: Int,
    @SerialName("columnAsientos")
    val columnAsientos: Int,
    @SerialName("precioEntrada")
    val precioEntrada: Double,
    val eventoTipo: EventoTipoDto,
    val integrantes: List<IntegranteDto>,
    val id: Long
) {
    @Serializable
    data class EventoTipoDto(
        val nombre: String,
        val descripcion: String
    )

    @Serializable
    data class IntegranteDto(
        val nombre: String,
        val apellido: String,
        val identificacion: String
    )
}
